### PR TITLE
Support for quoteStyle enforceQuoteStyle settings

### LINKED ISSUE #182: Formatting breaks attributes containing quotes
How to reproduce:
1. Use this document: `<hello attr='"' />`
2. Run the "Format document" command, with `xml.format.quotations` set to `doubleQuotes` (the default)

Symptoms:
The formatting produces the following invalid XML: `<hello attr=""" />`

Expected:
Formatting shouldn't change the type of quote used for an attribute if that produces invalid XML, that is if the attribute value contains the quote character.

### PR DESCRIPTION
**Pull request by xorye**
Fixes #182 

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

**Comment by xorye** [comment]
Please see https://github.com/eclipse/lemminx/pull/669

File: src/extension.ts [hunk 1]
```diff
@@ -188,6 +188,7 @@ export function activate(context: ExtensionContext) {
     xml['xml']['java']['home'] = javaHome;
     xml['xml']['format']['trimFinalNewlines'] = workspace.getConfiguration('files').get('trimFinalNewlines', false);
     xml['xml']['format']['insertFinalNewline'] = workspace.getConfiguration('files').get('insertFinalNewline', false);
+    xml['xml']['format']['quoteStyle'] = workspace.getConfiguration('xml').get('preferences.quoteStyle', "double");
```
Comment (xorye) [inline src/extension.ts:191]: The new preference for vscode-xml is `xml.preferences.quoteStyle`, but it is currently being mapped to `xml.format.quoteStyle` in order to make https://github.com/eclipse/lemminx/pull/669 simpler. 

Alternatively, I could remove this line, and on the lemminx side, create a new `XMLPreferences` class that contains `XMLPreferences#quoteStyle`.

File: README.md [hunk 2]
```diff
@@ -78,6 +78,8 @@ The following settings are supported:
 * `xml.symbols.excluded`: Disable document symbols (Outline) for the given file name patterns. Updating file name patterns does not automatically reload the Outline view for the relevant file(s). Each file must either be reopened or changed, in order to trigger an Outline view reload.
 
 Since 0.12.0:
+* `xml.format.enforceQuoteStyle`: Enable/disable enforce quote style when formatting. Default is `true`.
```
Comment (fbricon) [inline README.md:81]: xml.format.quotes: preferred|ignore : Enforce preferred quote style / ignore quote formatting

File: src/extension.ts (hunk 1 above)
Comment (fbricon) [reply src/extension.ts:191]: I'd rather stay consistent and have the XMLPreferences node on the server side

File: README.md (hunk 2 above)
Comment (xorye) [reply README.md]: fixed

**Comment by xorye** [comment]
I've also added enum descriptions in VS Code: image

File: src/extension.ts (hunk 1 above)
Comment (xorye) [reply src/extension.ts:191]: Okay, just making sure before I commit to this, if I take that route, I would have to add XMLPreferences as a new parameter in places like:
https://github.com/eclipse/lemminx/blob/c06f71a737bf1f84f82145005acd12843c891ee1/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/ICodeActionParticipant.java#L25-L26

and 

https://github.com/eclipse/lemminx/blob/c06f71a737bf1f84f82145005acd12843c891ee1/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/XMLBuilder.java#L34

I know in general, it's preferred to have a lower number of parameters. Is it fine to add a parameter in this case?

Comment (fbricon) [reply src/extension.ts:191]: Yes new pref

File: README.md [hunk 3]
```diff
@@ -78,6 +78,8 @@ The following settings are supported:
 * `xml.symbols.excluded`: Disable document symbols (Outline) for the given file name patterns. Updating file name patterns does not automatically reload the Outline view for the relevant file(s). Each file must either be reopened or changed, in order to trigger an Outline view reload.
 
 Since 0.12.0:
+* `xml.format.enforceQuoteStyle`: preferred|ignore : Enforce preferred quote style / ignore quote formatting.
+* `xml.preferences.quoteStyle`: Preferred quote style to use for completion: `single` quotes, `double` quotes. Default is `double`.
```
Comment (fbricon) [inline README.md:82]: `xml.format.quotations` should be removed

File: src/extension.ts (hunk 1 above)
Comment (fbricon) [reply src/extension.ts:191]: and remove  `xml['xml']['format']['quoteStyle']`

**Pull request merged**